Twitter client enhancements

tharak123455 opened the pull request:
<!-- Use this template by filling in information and copy and pasting relevant items out of the html comments. -->

# Relates to:

Issue related to Twitter client enhancements and environment configuration
Risks

<!-- This risks section is to be filled out before final review and merge. -->

# Risks
Low - included new functionality of Twitter target users configuration and environment example file. No core functionality changes.
# Background
## What does this PR do?
Adds dynamic Twitter target users configuration through environment variables(agent can now interact with configured user's tweets without mentioning agent)
Updates .env.example with Twitter-related configuration examples
Improves error handling for wallet providers in Twitter client
Enhances Twitter interaction handling

## What kind of change is this?
Improvements (misc. changes to existing features)

<!--
Bug fixes (non-breaking change which fixes an issue)
Improvements (misc. changes to existing features)
Features (non-breaking change which adds functionality)
Updates (new versions of included code)
-->

<!-- This "Why" section is most relevant if there is no linked issue explaining why. If there is a related issue it might make sense to skip this why section. -->
<!--
## Why are we doing this? Any context or related work?
-->

# Documentation changes needed?
My changes require a change to the project documentation.
Updated .env.example with new Twitter configuration options
Documentation for Twitter target users configuration

<!-- Please show how you tested the PR. This will really help if the PR needs to be retested, and probably help the PR get merged quicker. -->

# Testing

## Where should a reviewer start?
Check the updated .env.example file for new Twitter configurations
Review changes in packages/client-twitter/src/interactions.ts

## Detailed testing steps
Copy .env.example to .env
Configure TWITTER_TARGET_USERS with comma-separated usernames
Run the Twitter client
Bot responds to configured target users
Error handling works for wallet provider issues
Twitter interactions are properly processed

<!-- If there is anything about the deploy, please make a note. -->

# Deploy Notes
New environment variable TWITTER_TARGET_USERS needs to be configured if using custom target users
No database changes required
No special deployment steps needed

## Discord username
tharakesh5545

meppsilon committed:
Add hyperbolic api to eliza

meppsilon committed:
Add doc files

meppsilon committed:
Clean up old code

meppsilon committed:
Fix model type alias

lalalune committed:
Merge branch 'farcaster-neynar-client' of https://github.com/sayangel/eliza into HEAD

lalalune committed:
add lock file

lalalune committed:
Merge branch 'main' of https://github.com/ai16z/eliza

lalalune committed:
Merge branch 'main-add-lockfile'

lalalune committed:
v0.1.5-alpha.5

actions-user committed:
chore: update changelog

snobbee committed:
ci: re-enable coverage report upload to Codecov in CI workflow

jkbrooks committed:
Merge pull request #880 from snobbee/ci/re-enable-codecov-reporting

fix: re-enable coverage report upload to Codecov in CI workflow

btspoony committed:
Merge branch 'main' into tbh/plugin-flow

btspoony committed:
fix: lock file

shakkernerd committed:
perf: disable building docs on build command

shakkernerd committed:
Merge pull request #884 from ai16z/build_command

chore: disable building docs on build command

lalalune committed:
Merge pull request #874 from btspoony/tbh/plugin-flow

feat: Add Flow Blockchain plugin

shakkernerd committed:
chore: enhance dev script, performance improvement and add help message

shakkernerd committed:
Merge pull request #887 from ai16z/dev_command

chore: enhance dev script, performance improvement and add help message

v1xingyue committed:
github image cicd

v1xingyue committed:
cicd when release

oxSaturn committed:
chore: update models for groq

https://console.groq.com/docs/deprecations

shakkernerd committed:
chore: added more help message to the important notice text.

shakkernerd committed:
Merge pull request #891 from ai16z/dev_command

chore: added more help message to the important notice text.

oxSaturn committed:
fix models test

shakkernerd committed:
chore: remove redundant dev:build command

mcp97 committed:
testing and getting advanced trading working

shakkernerd committed:
chore: add a delay to the help message

shakkernerd committed:
Merge pull request #892 from ai16z/dev_command

chore: improved dev command

btspoony committed:
feat: update provider output

YoungPhlo committed:
rename file correct date

YoungPhlo committed:
wdygdtw 4 notes + docs metadata

btspoony committed:
feat: improve prompt

0xaguspunk committed:
Pass env variables when setting up GOAT

0xaguspunk committed:
Pass env variables when setting up GOAT

0xaguspunk committed:
Update chain to base

0xaguspunk committed:
Update chain to base

golryang committed:
Consistent language for Community & Contact links

L-jasmine committed:
chore: add `USE_GAIANET_EMBEDDING` into `.env.example`

L-jasmine committed:
fix: gaianet_embedding default dimensions to 768

btspoony committed:
feat: update evm transaction call cdc file

cygaar committed:
fix: twitter actions not triggering

cygaar commented on packages/client-twitter/src/interactions.ts:
```diff
@@ -103,20 +108,88 @@ export class TwitterInteractionClient {
+  const TARGET_USERS = targetUsersStr 
+      ? targetUsersStr.split(',').map(u => u.trim())
+      : [
+          'VitalikButerin',    
```
Im not a fan of having a default list - if target users weren't provided then we should avoid this logic altogether

cygaar committed:
Merge pull request #903 from cygaar/improve_twitter_actions

fix: twitter actions not triggering

cygaar committed:
Merge pull request #890 from oxSaturn/patch-1

chore: update models for groq

L-jasmine committed:
feat: add gaianet default model and baseUrl

cygaar committed:
fix: evaluation json parsing

cygaar committed:
Merge pull request #907 from cygaar/fix_evaulation_parsing

fix: evaluation json parsing

HashWarlock committed:
docs: add tutorial for Eliza in TEE

HashWarlock committed:
docs: add overview

HashWarlock committed:
Merge branch 'main' into add-tee-mode

oxSaturn committed:
feat: allow users to configure models for groq

mcp97 committed:
get working

mcp97 committed:
Merge branch 'main' of https://github.com/ai16z/eliza into realitySpiral/coinabseAdvancedTrading

mcp97 committed:
pnpm lock from rerunning

mcp97 committed:
linter

madjin committed:
Merge pull request #895 from YoungPhlo/docs/community-streams

docs: Add What Did You Get Done This Week #4 summaries and timestamps

mcp97 committed:
actually fix linter :)

onur-saf committed:
MAX_TWEET_LENGTH env implementation

asianviking committed:
feat: create README_TH.md

sayangel committed:
chore: replace console log with eliza logger

sayangel committed:
chore: remove unnecessary comments

sayangel committed:
chore: log cleanup

sayangel committed:
feature: make farcaster polling configurable

sayangel committed:
feature: add farcaster dry run option

tharak123455 committed:
Update interactions.ts

tharak123455 committed:
Update .env.example

sayangel committed:
fix: update response logic and prompt

sayangel committed:
chore: update env.example with new farcaster configs

v1xingyue committed:
pnpm build have bugs with Dockerfile

L-jasmine committed:
Merge commit 'fb24df4f421a3e4c5a900ec1885c42fb3cbce31e' into perf/gaianet-config

krittawitk053 committed:
fix: update README_TH

primemeridiem committed:
chore(typo): fix typo TH readme

asianviking committed:
chore: fix typo

L-jasmine committed:
fix: Set the SMALL MEDIUM LARGE Gaianet model in the env.

onur-saf committed:
update DEFAULT_MAX_TWEET_LENGTH to 280

onur-saf committed:
update default twitterPostTemplate string

HashWarlock committed:
fix: resolve lint errors

mcp97 committed:
chore: deprecate text based way of generating JSON

lalalune committed:
Merge pull request #918 from asianviking/main

feat: create README_TH.md

lalalune committed:
Merge pull request #915 from second-state/perf/gaianet-config

feat: update gaianet config

jkbrooks committed:
Merge pull request #910 from oxSaturn/feat/configure-models-groq

feat: allow users to configure models for groq

jkbrooks committed:
Merge pull request #899 from golryang/patch-1

chore: Consistent language for Community & Contact link label

cygaar committed:
Merge pull request #920 from Sifchain/realitySpiral/deprecateGenerateObject

chore: deprecate text based way of generating JSON

jkbrooks committed:
Merge pull request #914 from sayangel/farcaster-client-improvements

fix: Farcater client cleanup and fixed response logic

jkbrooks committed:
Merge pull request #912 from onur-saf/main

MAX_TWEET_LENGTH env implementation

mcp97 committed:
add advanced trade folder

jkbrooks committed:
Merge pull request #725 from Sifchain/realitySpiral/coinabseAdvancedTrading

feat: implement advanced coinbase trading

mcp97 committed:
feat: add readContract functionality to Coinbase plugin

- Introduced a new action `readContractAction` to read data from deployed smart contracts using the Coinbase SDK.
- Added `ReadContractContent` interface and `ReadContractSchema` for validation.
- Updated templates to include a `readContractTemplate` for user guidance.
- Enhanced the `tokenContractPlugin` description to reflect the new reading capabilities.

mcp97 committed:
feat: add readContract functionality to Coinbase plugin

- Introduced a new action `readContractAction` to read data from deployed smart contracts using the Coinbase SDK.
- Added `ReadContractContent` interface and `ReadContractSchema` for validation.
- Updated templates to include a `readContractTemplate` for user guidance.
- Enhanced the `tokenContractPlugin` description to reflect the new reading capabilities.

boxhock committed:
Fix use of Heurist model env vars

boxhock committed:
Fix use of Heurist model env vars

mcp97 committed:
get read contract working :)

mcp97 committed:
get read contract working :)

mcp97 committed:
remove hardcoding

mcp97 committed:
remove hardcoding

mcp97 committed:
update format

mcp97 committed:
update format

mcp97 committed:
get invoking smart contract working

mcp97 committed:
get invoking smart contract working

Commit:
Add NanoGPT model provider

Commit:
Allow configuring NanoGPT models

HashWarlock committed:
Merge branch 'main' into add-tee-mode

Commit:
feat: config eternalai model from env

Commit:
feat: config eternalai model from env

btspoony committed:
feat: use generateObjectArray instead of generateObjectDEPRECATED

Commit:
feat: config eternalai model from env

Commit:
feat: config eternalai model from env

samwhosung committed:
feat: add dynamic watch paths for agent development

samwhosung committed:
feat: add dynamic watch paths for agent development

shakkernerd committed:
chore: variables naming consistency

shakkernerd committed:
chore: variables naming consistency

shakkernerd committed:
Merge pull request #931 from samuveth/samuv/fix-dev

feat: add dynamic watch paths for agent development

shakkernerd committed:
Merge pull request #924 from boxhock/fix-heurist-model-selection

fix: use of Heurist model env vars

oxSaturn committed:
fix: update quickstart and .env.example

oxSaturn committed:
fix: update quickstart and .env.example

oxSaturn committed:
add default

oxSaturn committed:
add default

shakkernerd committed:
Merge pull request #932 from oxSaturn/fix/update-docs

fix: update quickstart and .env.example

oxSaturn committed:
fix: update package name in faq

oxSaturn committed:
fix: update package name in faq

bmgalego committed:
add missing callback handler to runtime evaluate method

jkbrooks committed:
Merge pull request #923 from Sifchain/realitySpiral/coinbaseReadInvokeSmartContracts

feat: add readContract / invokeContract functionality to Coinbase plugin

bmgalego committed:
process all responses actions

bmgalego committed:
process all responses actions

xwxtwd committed:
feat: Supports uploading files to AWS S3.

xwxtwd committed:
feat: Supports uploading files to AWS S3.

meppsilon committed:
Merge branch 'main' into add-hyperbolic

Commit:
docs: add WSL installation guide

Commit:
docs: add WSL installation guide

0xCardiE committed:
add many other EVM chains configs

0xCardiE committed:
add many other EVM chains configs

bmgalego committed:
fix response memory userId to agentId

bmgalego committed:
fix response memory userId to agentId

0xCardiE committed:
add types

0xCardiE committed:
add types

0xCardiE committed:
add imports

0xCardiE committed:
add imports

0xCardiE committed:
use proper names

0xCardiE committed:
use proper names

0xCardiE committed:
refactor files for easier maintance

0xCardiE committed:
refactor files for easier maintance

0xCardiE committed:
add other chains to templates

0xCardiE committed:
add other chains to templates

lalalune committed:
Merge pull request #948 from bmgalego/fix-telegram-response

fix: telegram response memory userId to agentId

0xCardiE committed:
fix the path to getChains

0xCardiE committed:
fix the path to getChains

Commit:
feat: config eternalai model from env

Commit:
feat: config eternalai model from env

odilitime committed:
feat: allow character.json settings => secrets to override models for OpenRouter

odilitime committed:
revert llamacloud endpoint change

odilitime committed:
rename intiface plugin

odilitime committed:
rename intiface-user-device-config.json

odilitime committed:
rename intiface plugin

odilitime committed:
rename intiface plugin

asianviking committed:
feat: add dev script to plugin-aptos

asianviking committed:
feat: add dev script to plugin-aptos

jkbrooks committed:
Merge pull request #927 from TrustlessComputer/pull-request

feat: Config eternalai model from env

jkbrooks committed:
Merge pull request #828 from meppsilon/add-hyperbolic

feat: add hyperbolic api to eliza

jkbrooks committed:
Merge pull request #946 from ileana-pr/add-wsl-guide

docs: add WSL installation guide

mcp97 committed:
Revert "docs: add WSL installation guide"

mcp97 committed:
Revert "docs: add WSL installation guide"

jkbrooks committed:
Merge pull request #959 from monilpat/revert-946-add-wsl-guide

fix: Revert "docs: add WSL installation guide"

oxSaturn committed:
fix: use MAX_TWEET_LENGTH from setting

oxSaturn committed:
fix: use MAX_TWEET_LENGTH from setting

oxSaturn committed:
use DEFAULT_MAX_TWEET_LENGTH

oxSaturn committed:
use DEFAULT_MAX_TWEET_LENGTH

btspoony committed:
Merge branch 'main' into tbh/flow-update-generateObject

0xAsten committed:
fix(agent): correct EVM plugin activation condition

The EVM plugin should be activated when the wallet address starts with '0x'

0xAsten committed:
fix(agent): correct EVM plugin activation condition

The EVM plugin should be activated when the wallet address starts with '0x'

sin-bufan committed:
fix: farcaster client does NOT fire action

sin-bufan committed:
fix: farcaster client does NOT fire action

sin-bufan committed:
add cast hash to memory in processActions

sin-bufan committed:
add cast hash to memory in processActions

HashWarlock committed:
fix: remove undefined function, add lint to plugin-tee

bkellgren committed:
client-linkedin initial

odilitime committed:
Merge pull request #963 from sin-bufan/fix-farcaster-client-processActiona

Fix farcaster client process action issue

odilitime committed:
Merge pull request #962 from 0xAsten/patch-1

fix(agent): correct EVM plugin activation condition

odilitime committed:
Merge pull request #960 from oxSaturn/fix/MAX_TWEET_LENGTH

fix: use MAX_TWEET_LENGTH from setting

odilitime committed:
Merge pull request #941 from xwxtwd/feat/support-aws-s3

feat: Supports upload files to AWS S3.

odilitime committed:
Merge pull request #937 from oxSaturn/fix/fix-faq

fix: update package name in faq

odilitime committed:
Merge pull request #940 from bmgalego/process-all-actions

feat: process all responses actions

odilitime committed:
Merge pull request #949 from n00b21337/947-add_other_EVMs

947 add other evm chains to wallet

odilitime committed:
Merge pull request #956 from asianviking/main

feat: add dev script to plugin-aptos

meppsilon committed:
Add hyperbolic env vars to override model class

meppsilon committed:
Add hyperbolic env vars to override model class

odilitime committed:
Merge pull request #974 from meppsilon/add-hyperbolic-vars

feat: Add hyperbolic env vars to override model class

odilitime committed:
Merge pull request #898 from goat-sdk/plugin-secrets

chore: pass env variables when setting up GOAT and update GOAT readme

yodamaster726 committed:
chore: add how to startup chat ui

yodamaster726 committed:
fix: remove copy file

odilitime committed:
Merge branch 'main' into add-nano-gpt-provider

odilitime committed:
Remove duplicate TOGETHER on switch

bmgalego committed:
fix jsdoc

shakkernerd committed:
Merge branch 'Phala-Network-add-tee-mode'

shakkernerd committed:
chore: fix broken lockfile

shakkernerd committed:
Merge pull request #977 from ai16z/fix/pnpm-lockfile

chore: fix broken lockfile

jkbrooks committed:
Merge pull request #954 from odilitime/fix-llamacloud

fix: revert llamacloud endpoint change

shakkernerd committed:
Merge pull request #938 from bmgalego/evaluator-callback

feat: add callback handler to runtime evaluate method

odilitime committed:
Merge branch 'main' into rename-intiface

odilitime committed:
rename package name

odilitime committed:
Merge branch 'rename-intiface' of https://github.com/odilitime/eliza into rename-intiface

odilitime committed:
add build-docker action

odilitime committed:
use build-docker

shakkernerd committed:
Merge pull request #978 from odilitime/fix-docker

fix: docker trying to filter out missing docs package

shakkernerd committed:
Merge pull request #955 from odilitime/rename-intiface

chore: rename intiface plugin

shakkernerd committed:
Merge pull request #953 from odilitime/fix-openrouter

feat: allow character.json settings models for open router

lalalune committed:
Merge pull request #973 from bkellgren/main

LinkedIn Client

v1xingyue committed:
modify build-docker

yodamaster726 committed:
fix: remove python reference

yodamaster726 committed:
chore: update readme to show how to how to run client ui and update package.json to start ui client

yodamaster726 committed:
Merge branch 'main' into chat-doc

lalalune committed:
build fix

lalalune committed:
Fix build

lalalune committed:
uhh fix linkedin lint lol

Commit:
updated step 3 and added code to make guide visible in docs page

btspoony committed:
Merge branch 'main' into tbh/flow-update-generateObject

jnaulty committed:
feat: make NodePlugin and GoatPlugin creation conditional

- NodePlugin is now only created when all required AWS credentials are present
  (AWS_ACCESS_KEY_ID, AWS_SECRET_ACCESS_KEY, AWS_REGION, AWS_S3_BUCKET)
- GoatPlugin is now only created when ALCHEMY_API_KEY is present
- Added type annotations for plugins to handle undefined cases

These changes prevent unnecessary plugin initialization when required
credentials are missing and improve type safety.

Issues introduced in: https://github.com/ai16z/eliza/pull/941 and
https://github.com/ai16z/eliza/pull/898

jnaulty committed:
do not activate aws s3 service in plugin-node if env vars not present

oxSaturn committed:
fix: a typo in characterfile.md

lalalune committed:
Merge pull request #986 from oxSaturn/fix/typo-in-characterfile

fix: a typo in characterfile.md

yodamaster726 committed:
fix: update text

dontAskVI committed:
Fix search for TwitterSearchClient

dontAskVI committed:
Remove .getInstance() from serach.ts

dontAskVI committed:
Comment search function out

dontAskVI committed:
Add image model for Galadriel model provider

odilitime committed:
Merge pull request #985 from jnaulty/jnaulty/run-without-alchemy-and-aws-env

fix: Goat Plugin + AWS S3 Service error when env vars absent

dontAskVI committed:
Merge branch 'main' into add-galadriel-img-gen-and-fix-search

mcp97 committed:
Merge pull request #983 from ileana-pr/add-wsl-guide

docs: add WSL Setup Guide to documentation

odilitime committed:
Merge pull request #976 from yodamaster726/chat-doc

chore: add how to startup chat ui

odilitime committed:
Merge pull request #929 from btspoony/tbh/flow-update-generateObject

feat: flow update generate object

odilitime committed:
Merge pull request #889 from v1xingyue/v1xingyue-patch-1

Feat : github image cicd

odilitime committed:
Merge pull request #926 from dylan1951/add-nano-gpt-provider

feat: Add NanoGPT provider

dontAskVI committed:
Add Galadriel comment to ModelProviderName.md

dontAskVI committed:
Clean up before PR

dontAskVI committed:
merge

dontAskVI committed:
Add pnpm-locl.yaml

GottliebFreudenreich committed:
Add files via upload

odilitime committed:
Merge pull request #994 from dontAskVI/add-galadriel-img-gen-and-fix-search

fix: Fix Twitter Search Logic and Add Galadriel Image Model

mcp97 committed:
Merge pull request #995 from GottliebFreudenreich/main

feat: create README_DE.md

ai16z-demirix committed:
test: adding parsing tests. changed files parsing.test.ts

mcp97 committed:
Merge pull request #996 from ai16z-demirix/main

oxSaturn committed:
feat: allow users to configure models for openai and anthropic

Commit:
Fix typo: initialize

mcp97 committed:
Merge pull request #999 from oxSaturn/feat/configure-models-openai-anthropic

feat: allow users to configure models for openai and anthropic

mcp97 committed:
Merge pull request #1000 from cryptofish7/fix-typo-initializeCache

fix: typo initialize

sin-bufan committed:
add callback to processActions in farcaster client

shakkernerd committed:
Merge pull request #1002 from sin-bufan/feat-farcaster-client-action-callback

fix: add callback to action in farcaster client

twilwa committed:
no more merging to main

must be from develop or tests will fail, failing tests will not allow
merge

tharak123455 committed:
Update interactions.ts

tharak123455 committed:
Merge branch 'main' into twitterClient-enhancements

tharak123455 committed:
Update interactions.ts
